tests/functional/common.js

```diff
@@ -0,0 +1,126 @@
+/* @flow */
+import {ChildProcess, spawn} from 'child_process';
+import copyDir from 'copy-dir';
+import path from 'path';
+import promisify from 'es6-promisify';
+
+
+import * as tmpDirUtils from '../../src/util/temp-dir';
+export const withTempDir = tmpDirUtils.withTempDir;
+
+
+export const functionalTestsDir = path.resolve(__dirname);
+export const projectDir = path.join(functionalTestsDir, '..', '..');
+export const webExt = path.join(projectDir, 'bin', 'web-ext');
+export const fixturesDir = path.join(functionalTestsDir, '..', 'fixtures');
+export const addonPath = path.join(fixturesDir, 'minimal-web-ext');
+export const fakeFirefoxPath = path.join(
+  functionalTestsDir, 'fake-firefox-binary.js'
+);
+export const fakeServerPath = path.join(
+  functionalTestsDir, 'fake-amo-server.js'
+);
+
+
+// withTempAddonDir helper
+
+export type TempAddonParams = {
+  addonPath: string,
+  runFromCwd?: boolean
+};
+
+export type TempAddonCallback =
+  (tmpAddonDir: string, tmpDir: string) => Promise<any>
+
+const copyDirAsPromised = promisify(copyDir);
+export function withTempAddonDir(
+  {addonPath, runFromCwd}: TempAddonParams,
+  makePromise: TempAddonCallback,
+): Promise<any> {
+  return withTempDir((tmpDir) => {
+    const tempAddonDir = path.join(tmpDir.path(), 'tmp-addon-dir');
+    return copyDirAsPromised(addonPath, tempAddonDir)
+      .then(() => {
+        // NOTE: when "web-ext run" is spawned in functional tests,
```

Comment:
@kumar303 This was a very weird issue that I still want to figure out (but in the meantime I workarounded it here):

if I change the current work dir of the functional test suite process, every tested web-ext command behaves as expected, but "web-ext run" doesn't understand "--no-reload" and it exits with the error message "Unknown argument: reload" (raised by yargs).

By tracing the parsed yargs argv (in both the cases: when the options are parsed correctly, and when yargs raises the above me error message), it seems that the issue is that yargs turns "--no-reload" into "reload: false" instead of "noReload: true".